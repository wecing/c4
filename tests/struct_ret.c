extern int printf(const char *s, ...);

typedef union U {
  long x;
  float y;
} U;

U f1(int n) {
  U u;
  u.x = n;
  return u;
}

typedef struct S {
  long x;
  float y;
  long z;
} S;

S f2(int n) {
  S s;
  s.x = n;
  return s;
}

int main() {
  if (1) {
    U u;
    u = f1(42);
    printf("u.x = %ld\n", u.x);
  }
  if (1) {
    S s;
    s = f2(99);
    printf("s.x = %ld\n", s.x);
  }
  return 0;
}
