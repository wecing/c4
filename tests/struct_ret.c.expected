u.x = 42
s.x = 99
